packages/shared_preferences/android/src/main/java/io/flutter/plugins/sharedpreferences/SharedPreferencesPlugin.java

```diff
@@ -88,6 +89,9 @@ private String encodeList(List<String> list) throws IOException {
           } else if (stringValue.startsWith(BIG_INTEGER_PREFIX)) {
             String encoded = stringValue.substring(BIG_INTEGER_PREFIX.length());
             value = new BigInteger(encoded, Character.MAX_RADIX);
+          } else if (stringValue.startsWith(DOUBLE_PREFIX)) {
```

Comment:
Please double check what happens to existing keys that are storing float values.

Reply:
Should I add the float check logic now?
I see there is 'setDouble' logic, no 'setFloat' logic, neither in iOS nor android.

Reply:
App which get updated need to be able to reuse the values that were saved with an older version of the app.

My comment was about making sure an app which used `setDouble() getDouble()` APIs work as intended before and after this change. If it does not, this is a breaking change and should be modified to not break the values saved in the older version.

My gut feeling is that it is not a breaking change since if you read Float back then you will return it to Flutter as a double. However, it would be nice to double check to be sure.

Reply:
I've double checked the old existing value will remain after this patch.
However, flutter/flutter#27222 will remain for those values.